stdlib/importlib/resources/simple.pyi

```diff
@@ -28,11 +28,31 @@ if sys.version_info >= (3, 11):
         def is_file(self) -> Literal[True]: ...
         def is_dir(self) -> Literal[False]: ...
         @overload
-        def open(self, mode: OpenTextMode = "r", *args: Incomplete, **kwargs: Incomplete) -> TextIOWrapper: ...
+        def open(
+            self,
+            mode: OpenTextMode = "r",
+            *,
+            encoding: str | None = ...,
+            errors: str | None = ...,
+            newline: str | None = ...,
+            line_buffering: bool = ...,
+            write_through: bool = ...,
+        ) -> TextIOWrapper: ...
         @overload
-        def open(self, mode: OpenBinaryMode, *args: Unused, **kwargs: Unused) -> BinaryIO: ...
+        def open(
+            self,
+            mode: OpenTextMode,
+            buffer: _WrappedBuffer,
+            encoding: str | None = ...,
+            errors: str | None = ...,
+            newline: str | None = ...,
+            line_buffering: bool = ...,
+            write_through: bool = ...,
+        ) -> TextIOWrapper: ...
         @overload
-        def open(self, mode: str, *args: Incomplete, **kwargs: Incomplete) -> IO[Any]: ...
+        def open(self, mode: OpenBinaryMode) -> BinaryIO: ...
```

Comment:
While args and kwargs are just ignored in this case, passing any further arguments is potentially a bug, which is why I removed the arguments and their `Unused` annotations.